proposals/lip-0045.md

```diff
@@ -984,166 +961,358 @@ The Interoperability module contains the following commands:
 
-    return terminateChainInternal(chainID)
+The `code` property of the data schema can take any of the following values:
+- `CCM_SENT_CODE_CHANNEL_UNAVAILABLE`
+- `CCM_SENT_CODE_MESSAGE_FEE_EXCEPTION`
+- `CCM_SENT_CODE_INVALID_FORMAT`
+
+##### Topics
+
+The event just has the default topic.
+
+##### Data
+
+```java
+ccmSentDataSchema = {
+    "type": "object",
+    "required": ["ccm", "code"],
+    "properties": {
+        "ccm": {
+           .. crossChainMessageSchema
+           "fieldNumber": 1
+        }
+        "code": {
+            "dataType": "uint32",
+            "fieldNumber": 2
+        }
+    }
+}
 ```
 
-#### getChainAccount
+#### TerminatedStateAccountCreated
 
-The `getChainAccount` function returns the object stored in the chain data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_STATE_CREATED`. This event is emitted when a new entry is created in the terminated state substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated chain account.
 
-##### Returns
+##### Data
 
-This function returns `chainAccount(chainID)`.
+The data property follows the `terminatedStateAccountSchema` schema given [above](#json-schema-5).
 
-#### getChannel
+#### TerminatedOutboxCreated
 
-The `getChannel` function returns the object stored in the channel data substore with store key equal to the input `chainID`.
+This event has `name = EVENT_NAME_TERMINATED_OUTBOX_CREATED`. This event is emitted when a new entry is created in the terminated outbox substore.
 
-##### Parameters
+##### Topics
 
-* `chainID`: The ID of the chain.
+- `chainID`: The ID of the terminated outbox account.
 
-##### Returns
+##### Data
 
-This function returns `channel(chainID)`.
+The data property follows the `terminatedOutboxAccountSchema` schema given [above](#json-schema-6).
 
+### Protocol Logic for Other Modules
 
-#### getOwnChainAccount
+#### send
 
-The `getOwnChainAccount` function returns the object stored in the own chain data substore.
+The `send` function is used to create and add a message to the outbox of a partner chain.
 
-##### Parameters
+```python
+def send(sendingAddress: Address,
+    module: Module,
+    crossChainCommand: CrossChainCommand,
+    receivingChainID: ChainID,
+    fee: uint64,
+    status: uint32,
+    params: object
+) -> None:
+
+    paramsSchema = JSON schema associated with (module, crossChainCommand)
+    # This could fail
+    params = encode(paramsSchema, params)
 
-This function takes no input.
+    ccm = {
+        "nonce": ownChainAccount.nonce,
+        "module": module,
+        "crossChainCommand": crossChainCommand,
+        "sendingChainID": ownChainAccount.chainID,
+        "receivingChainID": receivingChainID,
+        "fee": fee,
+        "status": status,
+        "params": params
+    }
 
-##### Returns
+    ccmID = sha256(encode(crossChainMessageSchema, ccm))
```

Comment:
I would move this line further below (just before it is needed), in particular, so that we run ` validateFormat(ccm)` before calling encode here.

Reply:
Sounds good. I thought that this was already fixed, but perhaps broken again at some point.

Reply:
Fixed.